zebrad/tests/config.rs

```diff
@@ -0,0 +1,342 @@
+
+    assert_eq!(config.network.network.to_string(), "Testnet");
+    assert_eq!(
+        config.rpc.listen_addr.unwrap().to_string(),
+        "127.0.0.1:8232"
+    );
+    assert_eq!(
+        config.metrics.endpoint_addr.unwrap().to_string(),
+        "127.0.0.1:9999"
+    );
+}
+
+#[test]
+fn config_nonexistent_file_errors() {
+    let _env = EnvGuard::new();
+
+    let nonexistent_path = PathBuf::from("/this/path/does/not/exist.toml");
+    let result = ZebradConfig::load(Some(nonexistent_path));
+    assert!(result.is_err(), "Should fail to load nonexistent file");
+}
+
+// --- Environment variable precedence ---
+
+#[test]
+fn config_env_override_defaults() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_NETWORK__NETWORK", "Testnet");
+    env.set_var("ZEBRA_RPC__LISTEN_ADDR", "127.0.0.1:8232");
+
+    let config = ZebradConfig::load(None).expect("load config with env vars");
+
+    assert_eq!(config.network.network.to_string(), "Testnet");
+    assert_eq!(
+        config.rpc.listen_addr.unwrap().to_string(),
+        "127.0.0.1:8232"
+    );
+}
+
+#[test]
+fn config_env_override_file() {
+    let env = EnvGuard::new();
+
+    let temp_dir = TempDir::new().expect("create temp dir");
+    let config_path = temp_dir.path().join("test_config.toml");
+
+    let test_config = r#"
+[network]
+network = "Mainnet"
+
+[rpc]
+listen_addr = "127.0.0.1:8233"
+"#;
+
+    fs::write(&config_path, test_config).expect("write test config");
+
+    env.set_var("ZEBRA_NETWORK__NETWORK", "Testnet");
+    env.set_var("ZEBRA_RPC__LISTEN_ADDR", "127.0.0.1:8232");
+
+    let config = ZebradConfig::load(Some(config_path)).expect("load config");
+
+    assert_eq!(config.network.network.to_string(), "Testnet");
+    assert_eq!(
+        config.rpc.listen_addr.unwrap().to_string(),
+        "127.0.0.1:8232"
+    );
+}
+
+#[test]
+fn config_invalid_toml_errors() {
+    let _env = EnvGuard::new();
+
+    let temp_dir = TempDir::new().expect("create temp dir");
+    let config_path = temp_dir.path().join("invalid_config.toml");
+
+    let invalid_config = r#"
+[network
+network = "Testnet"
+"#;
+
+    fs::write(&config_path, invalid_config).expect("write invalid config");
+
+    let result = ZebradConfig::load(Some(config_path));
+    assert!(result.is_err(), "Should fail to load invalid TOML");
+}
+
+#[test]
+fn config_invalid_env_values_error() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_RPC__LISTEN_ADDR", "invalid_address");
+
+    let result = ZebradConfig::load(None);
+    assert!(
+        result.is_err(),
+        "Should fail with invalid RPC listen address"
+    );
+}
+
+#[test]
+fn config_nested_env_vars() {
+    let env = EnvGuard::new();
+
+    env.set_var("ZEBRA_TRACING__FILTER", "debug");
+
+    let config = ZebradConfig::load(None).expect("load config with nested env vars");
+
+    assert_eq!(config.tracing.filter.as_ref().unwrap(), "debug");
```

Comment:
Optional/Nitpick: It would be nice to assert that these are some values instead of calling `.unwrap()`.

```suggestion
    assert_eq!(config.tracing.filter.as_ref(), Some("debug"));
```